src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Filters/OperationOutcomeExceptionFilterTests.cs

```diff
@@ -217,6 +217,40 @@ public void GivenATransactionFailedExceptionException_WhenExecutingAnAction_Then
             ValidateOperationOutcome(new TransactionFailedException(), HttpStatusCode.InternalServerError);
         }
 
+        [Fact]
+        public void GivenAnUnrecognizedException_WhenExecutingAnAction_ThenNoResponseShouldBeCreated()
+        {
+            var filter = new OperationOutcomeExceptionFilterAttribute(_fhirRequestContextAccessor);
+
+            _context.Exception = new Exception();
+
+            filter.OnActionExecuted(_context);
+
+            Assert.False(_context.ExceptionHandled);
+            Assert.Null(_context.Result);
+        }
+
+        [Fact]
+        public void GivenAnUnrecognizedExceptionAndInnerException_WhenExecutingAnAction_ThenNoResponseShouldBeCreated()
+        {
+            var filter = new OperationOutcomeExceptionFilterAttribute(_fhirRequestContextAccessor);
+
+            var exception = new Exception(null, new Exception());
+            _context.Exception = exception;
+
+            filter.OnActionExecuted(_context);
+
+            Assert.False(_context.ExceptionHandled);
+            Assert.Null(_context.Result);
+            Assert.Same(exception, _context.Exception); // ensure state is restored
+        }
+
+        [Fact]
+        public void GivenARequestRateExceededExceptionAsAnInnerException_WhenExecutingAnAction_ThenTheResponseShouldBeAnOperationOutcome()
+        {
+            ValidateOperationOutcome(new Exception(null, new RequestRateExceededException(null)), HttpStatusCode.TooManyRequests);
+        }
```

Comment:
Is it worth having a test to show that the inner exception is only used if the outer exception couldn't be parsed?

Reply:
Makes sense.

Reply:
There's actually no recognized exception that has a constructor that takes an inner exception, so we currently can't get get into the situation to test (without reflection)